zyenyo-discord/src/main/java/dataStructures/LongestCommonSubstring.java

```diff
@@ -0,0 +1,41 @@
+package dataStructures;
+
+/**
+ * Finds the longest common substring present in the second string,
+ * from the first string.
+ * Note, this is NOT Longest Common Subsequence.
+ */
+public class LongestCommonSubstring
+{
+	public static String find(String str1, String str2)
+	{
+		int m = str1.length();
+		int n = str2.length();
```

Comment:
It would be nice to have more descriptive names than str1 and str2 here. For eg. `needle` and `haystack` could be better since it tells us which is being searched *for* and which is being searched *in*. I understand the comment states it, but readability never hurts.

I would also argue that single letter variables names make things difficult to read, but I understand if you needed to do it for conciseness. Changing this is up to your appreciation.

Reply:
Good point. B-)
Needle and haystack, I like it.